### DISCUSSION TITLE
Azure Pipeline YML to build Uno Android with .NET 8 RC2?

### DISCUSSION THREAD

**Discussion by VincentH-Net** [unoplatform/uno#13995] [Q&A]
Hi,

Are there any up to date Azure Pipeline YML examples / sources to build an Uno app for Android with .NET 8 RC2?

I'd also be happy with a version for .NET 7 that works today plus a few modification pointers.

I did find the [YML for Uno.Gallery](https://github.com/unoplatform/Uno.Gallery/blob/master/azure-pipelines.yml) but I'm unclear if that is the right starting point if I just want to build a signed Android App Bundle

Thanks!

**Comment by jeromelaban** [comment]
Did you find your answer since then? You should be able to build properly using `uno-check` with the `--pre-major` flag.

**Reply by VincentH-Net** [reply]
Thanks for the `uno-check` tip.

I did not find a good reference in a single place - had to piece YML together from various sources ([1](https://criticalhittech.com/2023/05/16/deploying-xamarin-and-maui-apps-from-azure-pipelines-to-google-play/), [2](https://marketplace.visualstudio.com/items?itemName=ms-vsclient.google-play&targetId=d78457db-963a-4987-87c4-6f606051309b#configuring-your-google-play-publisher-credentials), [3](https://learn.microsoft.com/en-us/appcenter/distribution/stores/googleplay)) plus ChatGpt.

However, functionality is quite minimal: 
- Build Android in release config
- Create APK and ABB packages
- Publish ABB to Google Play internal testing track

What would be really valuable and a great timesaver is a doc on the Uno site that covers how to support above in Azure (and Github?) plus:
- All platforms
- Run Uno UI Tests - possibly also "Golden" screenshot tests
- Deploy to Play store / Testflight / Azure static website / ...

If you already have pointers for above I'd really appreciate if you could share them here

**Reply by jeromelaban** [reply]
We have added more advanced CI pipelines in our default template from the wizard, this may help you get further along.
